packages/obonode/obojobo-sections-assessment/server/models/assessment-score.js

```diff
@@ -0,0 +1,158 @@
+				logger.error('AssessmentScore fetchById Error', error.message)
+				return Promise.reject(error)
+			})
+	}
+
+	static getImportableScore(userId, draftContentId, isPreview) {
+		return db
+			.oneOrNone(
+				`
+				SELECT
+					id,
+					created_at,
+					assessment_id,
+					score,
+					attempt_id
+				FROM assessment_scores
+				WHERE
+					user_id = $[userId]
+					AND draft_content_id = $[draftContentId]
+					AND is_preview = $[isPreview]
+					AND is_imported = false
+				ORDER BY
+					score DESC,
+					created_at DESC
+				LIMIT 1
+				`,
+				{
+					userId,
+					isPreview,
+					draftContentId
+				}
+			)
+			.then(result => {
+				if (result) {
+					return new AssessmentScore(result)
+				}
+			})
+			.catch(error => {
+				logger.error('AssessmentScore getImportableScore Error', error.message)
+				return Promise.reject(error)
+			})
+	}
+
+	static deletePreviewScores({ transaction, userId, draftId, resourceLinkId }) {
+		return transaction
+			.manyOrNone(
+				`
+				SELECT assessment_scores.id
+				FROM assessment_scores
+				JOIN attempts
+					ON attempts.id = assessment_scores.attempt_id
+				WHERE assessment_scores.user_id = $[userId]
+				AND assessment_scores.draft_id = $[draftId]
+				AND attempts.resource_link_id = $[resourceLinkId]
+				AND assessment_scores.is_preview = true
+			`,
+				{ userId, draftId, resourceLinkId }
+			)
+			.then(assessmentScoreIdsResult => {
+				const ids = assessmentScoreIdsResult.map(i => i.id)
+				if (ids.length < 1) return []
+
+				return [
+					transaction.none(
+						`
+						DELETE FROM lti_assessment_scores
+						WHERE assessment_score_id IN ($[ids:csv])
+					`,
+						{ ids }
+					),
+					transaction.none(
+						`
+						DELETE FROM assessment_scores
+						WHERE id IN ($[ids:csv])
+					`,
+						{ ids }
+					)
+				]
+			})
+	}
+
+	clone() {
+		const clone = Object.assign({}, this)
+		return new AssessmentScore(clone)
+	}
+
+	create(dbOrTransaction = db) {
+		if (this.id) throw Error('Cannot call create on a model that has an id.')
+		return dbOrTransaction
+			.one(
+				`
+			INSERT INTO assessment_scores
+				(user_id, draft_id, draft_content_id, assessment_id, attempt_id, score, score_details, is_preview, is_imported, imported_assessment_score_id)
+				VALUES($[userId], $[draftId], $[draftContentId], $[assessmentId], $[attemptId], $[score], $[scoreDetails], $[isPreview], $[isImported], $[importedAssessmentScoreId])
+				RETURNING id
+			`,
+				this
+			)
+			.then(result => {
+				this.id = result.id
+				return this
+			})
+	}
+
+	importAsNewScore(attemptId, resourceLinkId, dbTransaction) {
+		const newScore = this.clone()
+		delete newScore.id
+		newScore.attemptId = attemptId
+		newScore.isImported = true
+		newScore.importedAssessmentScoreId = this.id // use original item as imported id
+		newScore.resourceLinkId = resourceLinkId // update resourceLink
+		newScore.scoreDetails.attemptNumber = 1 // fix the attemptNumber
+		// dispatch an event?
```

Comment:
It may be important to fire a new event, maybe `assessment:scoreImported`?

Reply:
hmm.  check out https://github.com/iturgeon/Obojobo/blob/dev/12-ammolite-staged/packages/obonode/obojobo-sections-assessment/server/attempt-end/insert-events.js#L118-L166

Looks like I implemented a secondary event for 'assessment:attemptEnd' that has score import data in it.  Not sure what to do about it right now 🤔 

Although it does look like the eventVersions of those 2 paths to inserting 'assessment:attemptEnd' need to match up.

Reply:
Ahh okay - I'm thinking we probably should go with a new event here. If we leave it as `assessment:attemptEnd` then the properties in the payload between different `assessment:attemptEnd` events won't always be the same, and I like that the name of the event plus the version currently does guarantee that and don't want to break that consistency.

Reply:
I have no issue w/ that with the exception of importing may break old log processors.  I kinda want to store both events in the case of an import and just attemptEnd in the case of a normal completion

Reply:
That makes sense to me. Okay - we can add the import values to both events and in one they can just be set to null.

Reply:
Resolved with 961ce1d50604a6eefd5d361d1d96e8155aed6033